internal/batches/service/remote.go

```diff
@@ -98,13 +113,122 @@ func (svc *Service) CreateBatchSpecFromRaw(
 		"allowIgnored":     allowIgnored,
 		"allowUnsupported": allowUnsupported,
 		"noCache":          noCache,
+		"batchChange":      batchChange,
 	}).Do(ctx, &resp); err != nil || !ok {
 		return "", err
 	}
 
 	return resp.CreateBatchSpecFromRaw.ID, nil
 }
 
+// UploadMounts uploads file mounts to the server.
+func (svc *Service) UploadMounts(workingDir string, batchSpecID string, steps []batches.Step) error {
```

Comment:
when we land here, the existence of all files is already validated, correct?

Reply:
That's correct